src/utils/config.ts

```
@@ -139,6 +139,21 @@ export class ConfigClient {
       constants.isSupportedChainId(chainId) && this.spokeChains.has(chainId)
     );
   };
+  getSupportedCanonicalNameAsChainId = (canonicalName: string) => {
+    // Transform the canonical name to match ChainId key
+    const modifiedCanonicalName = canonicalName.toLowerCase();
+    // Attempt to resolve the chainId and return
+    const resolvedChain = constants.CanonicalChainName[modifiedCanonicalName];
+    return resolvedChain && this.isSupportedChainId(resolvedChain)
+      ? resolvedChain
+      : undefined;
+  };
+  resolveChainIdFromNumericOrCanonical = (chainIdOrCanonical?: string) => {
+    return chainIdOrCanonical
+      ? this.getSupportedCanonicalNameAsChainId(chainIdOrCanonical) ??
+          Number(chainIdOrCanonical)
+      : Number(chainIdOrCanonical);
```

if `chainIdOrCanonical` does not exist this ends up returning `NaN`. IMO i would just throw errors with invalid inputs or something that doesnt exist and handle that upstream, but i realize that is annoying in react